tests/components/weather/test_init.py

```diff
@@ -1340,3 +1341,115 @@ async def async_forecast_daily(self) -> list[Forecast] | None:
         "This is deprecated and will stop working in Home Assistant 2024.6. "
         "Use 'weather.get_forecasts' instead which supports multiple entities"
     ) in caplog.text
+
+
+@pytest.mark.parametrize(
+    (
+        "forecast_type",
+        "forecast_attribute",
+    ),
+    [
+        (
+            "daily",
+            ATTR_WEATHER_TEMPERATURE,
+        ),
+    ],
+)
+async def test_get_forecast_attr(
+    hass: HomeAssistant,
+    config_flow_fixture: None,
+    forecast_type: str,
+    forecast_attribute: str,
+) -> None:
+    """Test get forecast service."""
+
+    class MockWeatherMockForecast(MockWeatherTest):
+        """Mock weather class with mocked new method and legacy forecast."""
+
+        @property
+        def forecast(self) -> list[Forecast] | None:
+            """Return the forecast."""
+            return self.forecast_list
+
+        async def async_forecast_daily(self) -> list[Forecast] | None:
+            """Return the forecast_daily."""
+            return self.forecast_list
```

Comment:
Could we create a more proper forecast list here with more than 4 forecasts in it to use in both this and next test so we see that limit actually does what it should

Reply:
What is the best way to do it?